source/Server.Contracts/Variables/ActionSpecialVariable.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+
+namespace Sashimi.Server.Contracts.Variables
+{
+    public class ActionSpecialVariable
+    {
+        public ActionSpecialVariable(string name, BindingBehaviour userVisible = BindingBehaviour.Visible, params string[] deprecatedAliases)
+        {
+            Name = name;
+            DeprecatedAliases = deprecatedAliases;
+            BindingBehaviour = userVisible;
+        }
+
+        public string Name { get; }
+        public IEnumerable<string> DeprecatedAliases { get; }
+        public BindingBehaviour BindingBehaviour { get; }
+
+        public static implicit operator string?(ActionSpecialVariable typedString)
+        {
+            return typedString?.Name;
+        }
+
+        //Override ToString() because these used to be constants, so we want to retain this behaviour in case we are constructing a string somewhere
+        public override string ToString()
+        {
+            return Name;
+        }
+    }
+
+    public enum BindingBehaviour
```

Comment:
Should this just be a boolean instead? e.g. `IsVisibleToUsers`

Reply:
This is a bit like TinyTypes, having the enum makes the call very clear as to what is happening. `new SpecialVariable("foo", BindingBehaviour.Hidden)` is easier to reason about reading the calling code than `new SpecialVariable("foo", false)`

Reply:
hmm ... sounds like we are trying to get rid of any usage of boolean for better readability, am I right?
Is `Visibility` a better name than `BindingBehaviour` ?